### PR TITLE
[ADD] Create coding style report file when run nix command

### PR DESCRIPTION
**Pull request by TiluWeb**
When the command "nix run github:epitech/coding-style-checker" was run, the number of coding-style errors doubled.
This modification solves this problem and creates the report file "coding-style-reports.log".

### CODE REVIEW

**Comment by Sigmanificient** [comment]
Hello, I am the author of the script providing for Nix support.
It was added a long time ago, and my attempts at updating it were unsuccessful.

**TLDR**: *If you are using Nix to run the coding style, I highly recommend you use [this version instead](https://github.com/Sigmapitech/cs) which is up-to-date.*

Concerning the missing rule violation you reported, is it composed of 2 subproblems:
- The `/dev/stderr` pipe is incorrect, causing loss of the rule logging.
- The `flake.lock` file uses an old revision to fetch the ruleset repository content. It should be updated, but I am not able to maintain it properly as pull requests are not frequently read on this very repository.

Note that updating the flakes input with its current form, (using `nix flake update`) would break the report script, as it does not provide the correct runtime dependencies that were added a few months ago (on the ruleset, python requires `clang.index` within the python instance provided by banana-vera).

I made my version that addresses all these issues, and that I keep up to date.
It is located in this repository: https://github.com/Sigmapitech/cs.

However, concerning the report file creation, I believe that there is no real need for such functionality.
You can achieve a similar result by using shell to redirect the stderr & stdout and combine it with `tee`:
```bash
nix run github:Sigmapitech/cs . 2>&1 | tee report;
```

**Comment by TiluWeb** [comment]
Okay, thank you for the time you've taken to explain, have a good evening 👍
